src/domain/profile/Profile.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { FunctionComponent } from 'react';
+import { connect } from 'react-redux';
+import { useTranslation } from 'react-i18next';
+import { useHistory } from 'react-router-dom';
+import { useQuery } from '@apollo/react-hooks';
+import { gql } from 'apollo-boost';
+
+import NoUpcomingEvents from './components/noUpcomingEvents/NoUpcomingEvents';
+import { StoreState } from '../app/types/AppTypes';
+import { profileToStore } from './state/ProfileActions';
+import { GuardianValues } from './types/ProfileTypes';
+import { userSelector } from '../auth/state/AuthenticationSelectors';
+import { Children } from '../child/types/ChildTypes';
+import { normalizeChildren } from '../child/childUtils';
+
+const profileQuery = gql`
+  query profileQuery {
+    guardians {
+      edges {
+        node {
+          firstName
+          lastName
+          children {
+            edges {
+              node {
+                firstName
+                lastName
+                birthdate
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+`;
+interface Props {
+  profileToStore: (values: GuardianValues) => void;
+}
+
+const Profile: FunctionComponent<Props> = ({ profileToStore }) => {
+  const { loading, error, data } = useQuery(profileQuery);
+  const { t } = useTranslation();
+  const history = useHistory();
+
+  if (loading) return <div>'Loading...'</div>;
+  if (error) return <div>Error! {error.message}</div>;
+
+  /*
+   * If the guardian does not exist in the api, they have not registered and we want to send them to the front page.
+   */
+  if (data.guardians.edges.length === 0) {
+    history.push('/home');
+    return <div>No profile exists</div>;
+  } else {
+    const guardian = {
+      phone: data.guardians.edges[0].node.phone,
+      firstName: data.guardians.edges[0].node.firstName,
+      lastName: data.guardians.edges[0].node.lastName,
+    };
+
+    const children: Children = normalizeChildren(
+      data.guardians.edges[0].node.children.edges
+    );
+
+    const profileValues: GuardianValues = {
```

Comment:
You should do all this login in reducer, instead of exposing them here, look quite messy

Reply:
Send data response as action payload

Reply:
Chi